include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -81,20 +81,20 @@ namespace internal
     void
     get_element_type_specific_information(
       const FiniteElement<dim, dim> &fe_in,
-      const FiniteElement<dim, dim> *fe,
+      const FiniteElement<dim, dim> &fe,
       const unsigned int             base_element_number,
       ElementType &                  element_type,
       std::vector<unsigned int> &    scalar_lexicographic,
       std::vector<unsigned int> &    lexicographic_numbering)
     {
       element_type = tensor_general;
 
-      const auto fe_poly = dynamic_cast<const FE_Poly<dim, dim> *>(fe);
+      const auto fe_poly = dynamic_cast<const FE_Poly<dim, dim> *>(&fe);
```

Comment:
It should also work to cast as reference here.
```suggestion
      const auto& fe_poly = dynamic_cast<const FE_Poly<dim, dim> &>(fe);
```

Reply:
I'm not sure this will work - we later check if `fe_poly` is not `nullptr` so this would result in an exception in a condition where the code currently executes correctly.

Reply:
OK. I missed that we check for it being `nullptr`.

Reply:
Yes, correct: If the `dynamic_cast` fails for a pointer, you get a `nullptr`. If it fails for a reference, you get an exception -- that's not what we want.